fsspec/implementations/tests/test_arrow.py

```diff
@@ -0,0 +1,162 @@
+import secrets
+
+import pytest
+
+pyarrow_fs = pytest.importorskip("pyarrow.fs")
+FileSystem = pyarrow_fs.FileSystem
+
+from fsspec.implementations.arrow import ArrowFSWrapper  # noqa
+
+
+@pytest.fixture(scope="function")
+def fs():
+    fs, _ = FileSystem.from_uri("mock://")
+    return ArrowFSWrapper(fs)
+
+
+@pytest.fixture(scope="function")
+def remote_dir(fs):
+    directory = secrets.token_hex(16)
+    fs.makedirs(directory)
+    yield directory
+    fs.rm(directory, recursive=True)
+
+
+def strip_keys(original_entry):
+    entry = original_entry.copy()
+    entry.pop("mtime")
+    return entry
+
+
+def test_info(fs, remote_dir):
+    fs.touch(remote_dir + "/a.txt")
+    details = fs.info(remote_dir + "/a.txt")
+    assert details["type"] == "file"
+    assert details["name"] == remote_dir + "/a.txt"
+    assert details["size"] == 0
+
+    fs.mkdir(remote_dir + "/dir")
+    details = fs.info(remote_dir + "/dir")
+    assert details["type"] == "directory"
+    assert details["name"] == remote_dir + "/dir"
+
+    details = fs.info(remote_dir + "/dir/")
+    assert details["name"] == remote_dir + "/dir/"
+
+
+def test_move(fs, remote_dir):
+    fs.touch(remote_dir + "/a.txt")
+    initial_info = fs.info(remote_dir + "/a.txt")
+
+    fs.move(remote_dir + "/a.txt", remote_dir + "/b.txt")
+    secondary_info = fs.info(remote_dir + "/b.txt")
+
+    assert not fs.exists(remote_dir + "/a.txt")
+    assert fs.exists(remote_dir + "/b.txt")
+
+    initial_info.pop("name")
+    secondary_info.pop("name")
+    assert initial_info == secondary_info
```

Comment:
You might also want to add a test moving to a directory (with recursive=True), see eg https://github.com/intake/filesystem_spec/blob/0589358d8a029ed6b60d031018f52be2eb721291/fsspec/implementations/tests/test_local.py#L622

Reply:
Sure, done!